### PR TITLE
Kapt: Add flag to keep KDoc comments in stubs

### PR DESCRIPTION
**Pull request by hungvietnguyen**

### CODE REVIEW

File: libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/internal/kapt/Kapt3KotlinGradleSubplugin.kt [hunk 1]
```diff
@@ -386,12 +405,14 @@ class Kapt3GradleSubplugin @Inject internal constructor(private val registry: To
             project.logger.warn("'kapt.generateStubs' is not used by the 'kotlin-kapt' plugin")
         }
 
+        // These option names must match those defined in `org.jetbrains.kotlin.kapt.cli.KaptCliOption`.
         pluginOptions += SubpluginOption("useLightAnalysis", "${kaptExtension.useLightAnalysis}")
         pluginOptions += SubpluginOption("correctErrorTypes", "${kaptExtension.correctErrorTypes}")
         pluginOptions += SubpluginOption("dumpDefaultParameterValues", "${kaptExtension.dumpDefaultParameterValues}")
         pluginOptions += SubpluginOption("mapDiagnosticLocations", "${kaptExtension.mapDiagnosticLocations}")
-        pluginOptions += SubpluginOption("strictMode", "${kaptExtension.strictMode}")
+        pluginOptions += SubpluginOption("strictMode", "${kaptExtension.strictMode}") // Currently doesn't match KaptCliOption, not sure if it's intended.
```
Comment (hungvietnguyen) [inline libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/internal/kapt/Kapt3KotlinGradleSubplugin.kt:413]: Possible typo detected: Here this option name is `strictMode`; but in `KaptCliOption`, this option name is `strict`. (They were first introduced in https://github.com/JetBrains/kotlin/commit/c83581e6b8155c6d89da977be6e3cd4af30562e5.) // cc @yanex

File: libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/internal/kapt/Kapt3KotlinGradleSubplugin.kt [hunk 2]
```diff
@@ -74,11 +74,14 @@ class Kapt3GradleSubplugin @Inject internal constructor(private val registry: To
...
-        private val VERBOSE_OPTION_NAME = "kapt.verbose"
-        private val USE_WORKER_API = "kapt.use.worker.api"
-        private val INFO_AS_WARNINGS = "kapt.info.as.warnings"
-        private val INCLUDE_COMPILE_CLASSPATH = "kapt.include.compile.classpath"
-        private val INCREMENTAL_APT = "kapt.incremental.apt"
+        // Pairs of property names and their default values.
+        // The default values of these properties should typically match those defined in org.jetbrains.kotlin.base.kapt3.KaptFlag,
+        private val VERBOSE_OPTION_NAME = Pair("kapt.verbose", false)
+        private val USE_WORKER_API = Pair("kapt.use.worker.api", true)
+        private val INCREMENTAL_APT = Pair("kapt.incremental.apt", true) // Currently doesn't match KaptFlag, not sure if it's intended.
```
Comment (hungvietnguyen) [inline libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/internal/kapt/Kapt3KotlinGradleSubplugin.kt:81]: Inconsistency detected: This flag is disabled by default according to the code at `org.jetbrains.kotlin.base.kapt3.KaptOptions.Builder#getFlags`--see below. // cc @gavra0

```
        // This set includes enabled-by-default flags. It doesn't contain INCREMENTAL_APT.
        val flags: MutableSet<KaptFlag> = mutableSetOf(
            KaptFlag.USE_LIGHT_ANALYSIS,
            KaptFlag.INCLUDE_COMPILE_CLASSPATH
        )
```

**Review by gavra0 — APPROVED** [review]
Look good, thanks!

File: libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/internal/kapt/Kapt3KotlinGradleSubplugin.kt (hunk 2 above)
Comment (hungvietnguyen) [reply libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/internal/kapt/Kapt3KotlinGradleSubplugin.kt]: Cool, I've added a commit to add a comment for that: https://github.com/JetBrains/kotlin/pull/3942/commits/571b63ac10fa73c8c33c622928dc77c76d48abd5

(@nav-nav Nataliya told me she prefers avoiding force pushes, and will help with squashing commits.)

**Comment by hungvietnguyen** [comment]
@nav-nav Hi Nataliya, as discussed I have split this PR into 2 to make it easier to review. This PR now has 2 commits: the first one is the main commit, and the second ones fixes the failed tests.

I'll upload the second PR after this PR gets merged.

**Comment by hungvietnguyen** [comment]
@nav-nav Hi Nataliya, the updated commit has passed all the tests in TeamCity, including the `(IR)ClassFileToSourceStubConverterTestGenerated.testCommentsRemoved` tests that were failing before: https://teamcity.jetbrains.com/test/-3890362769634311212?currentProjectId=Kotlin_KotlinForGoogle_Tests&branch=prr%2Fhungnv%2FKT-43593-addFlagToKeepKDocCommentsInStubs

I had to force-push the branch because it was far behind tip-of-tree. The main change I made to fix the failed tests is at [line 264 of AbstractClassFileToSourceStubConverterTest](https://github.com/JetBrains/kotlin/pull/3942/commits/6011be1e4b952a097fe7db0d33c166725d57d2d7#diff-e7d1a515345521203d5f15532f7421948cf1fb4cc87a58f013d59ca33ca85476R264) and [line 1 of commentsRemoved.kt](https://github.com/JetBrains/kotlin/pull/3942/commits/6011be1e4b952a097fe7db0d33c166725d57d2d7#diff-961014c57ab7754ce7d2efb3434fd55c665a49f857a8e9f55144e592aecfb931R1).

Could you help verify this PR again? Thanks!

File: plugins/kapt3/kapt3-compiler/test/org/jetbrains/kotlin/kapt3/test/AbstractKotlinKapt3Test.kt [hunk 3]
```diff
@@ -251,30 +252,25 @@ open class AbstractClassFileToSourceStubConverterTest : AbstractKotlinKapt3Test(
...
         super.doTest(filePath)
         doTestWithJdk9(AbstractClassFileToSourceStubConverterTest::class.java, filePath)
         doTestWithJdk11(AbstractClassFileToSourceStubConverterTest::class.java, filePath)
     }
 
     override fun check(kaptContext: KaptContextForStubGeneration, javaFiles: List<File>, txtFile: File, wholeFile: File) {
-        val generateNonExistentClass = wholeFile.isOptionSet("NON_EXISTENT_CLASS")
-        val validate = !wholeFile.isOptionSet("NO_VALIDATION")
+        val generateNonExistentClass = isFlagEnabled("NON_EXISTENT_CLASS", wholeFile)
+        val validate = !isFlagEnabled("NO_VALIDATION", wholeFile)
```
Comment (nav-nav) [inline plugins/kapt3/kapt3-compiler/test/org/jetbrains/kotlin/kapt3/test/AbstractKotlinKapt3Test.kt:273]: It seems you have created isFlagDisabled for this case.

**Comment by nav-nav** [comment]
Merged into master

File: plugins/kapt3/kapt3-compiler/test/org/jetbrains/kotlin/kapt3/test/AbstractKotlinKapt3Test.kt (hunk 3 above)
Comment (hungvietnguyen) [reply plugins/kapt3/kapt3-compiler/test/org/jetbrains/kotlin/kapt3/test/AbstractKotlinKapt3Test.kt:273]: Yes, but `isFlagDisabled` is not the same as`!isFlagEnabled`. `isFlagDisabled` checks the existence of the string `// !{flagName}` in a test file, whereas `!isFlagEnabled` checks the absence of the string `// {flagName}`. Here the existing code wants to use `!isFlagEnabled`, so I'm trying to keep that behavior.

Thank you for merging this PR!

**Pull request closed without merging**